File: src/subplot_example.cpp
#include "../include/matplotlib.h"

#include <iostream>
#include <vector>
#include <cmath>


namespace plt = matplotlibcpp;

int main() {

    const int n = 100;
    std::vector<double> x(n), sinY(n), cosY(n);

    for (int i = 0; i < n; ++i) {
        x[i] = i * 2 * M_PI / (n - 1);
        sinY[i] = std::sin(x[i]);
        cosY[i] = std::cos(x[i]);
    }

    
    plt::subplot(1, 2, 1);
    plt::plot(x, sinY, "b");
    plt::title("Sine");
    plt::grid(true);

    plt::subplot(1, 2, 2);
    plt::plot(x, cosY, "r");
    plt::title("Cosine");
    plt::grid(true);

    plt::tight_layout();
    plt::show();

    return 0;
}

File: py_examples/subplot_example.py
import matplotlib.pyplot as plt
import numpy as np

n = 100
x = np.linspace(0, 2 * np.pi, n)
sinY = np.sin(x)
cosY = np.cos(x)

plt.subplot(1, 2, 1)
plt.plot(x, sinY, 'b')
plt.title("Sine")
plt.grid(True)

plt.subplot(1, 2, 2)
plt.plot(x, cosY, 'r')
plt.title("Cosine")
plt.grid(True)

plt.tight_layout()
plt.show()